modules/gateway/gateway_test.go

```diff
@@ -6,21 +6,27 @@ import (
 
 	"github.com/NebulousLabs/Sia/consensus"
 	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/modules/tester"
 )
 
-var port = 9001
+var (
+	rpcPort = 10000
+)
 
-func newTestGateway(t *testing.T) *Gateway {
-	g, err := New(":"+strconv.Itoa(port), consensus.CreateGenesisState(), "")
+// newTestingGateway returns a gateway read to use in a testing environment.
+func newTestingGateway(directory string, t *testing.T) *Gateway {
+	gDir := tester.TempDir(directory, modules.GatewayDir)
```

Comment:
also gives you a little less control. Some functions need multiple. (Like the daemon testing peer stuff uses multiple)